### PR TITLE
chore: extend namespace globally for request properties and log info utility function

### LINKED ISSUE #415: extend namespace globally for request properties
Related to #405

Currently the Request interface does not contain properties such as id. This should be extended globally for consistency throughout the application

### PR DESCRIPTION
**Pull request by shuli-ogp**
## Problem

Closes #415 and #416

## Solution
- create getMeta() utility function to extract relevant log info (i.e. ip, trace, url, header) instead of specifying at each logger function call
- extend Request interface globally

## Tests

**On staging**
- [ ] At the login page, key in an invalid email address abc@abc.com. Check that the value of cf-ray header is logged under trace in the server logs error message.
- [ ] Access a public form. Check that the value of cf-ray header is logged under trace in the server logs for the get request.

**On localhost**
- [ ] At the login page, key in an invalid email address abc@abc.com. Check that the value of x-request-id header is logged under trace in the server logs error message.
- [ ] In L25-L27 of `logging.ts`, replace
```
isDev
  ? format.combine(format.colorize(), errorPrinter(), customFormat)
  : format.json({ replacer: jsonErrorReplacer }),
```
with 
``` 
format.json({ replacer: jsonErrorReplacer }),
```
Access a public form. Check that the value of x-request-id header is logged under trace in the server logs for the get request.

### CODE REVIEW

**Comment by shuli-ogp** [comment]
@karrui 
Re your comment in #405 that we should extend `request` to also have `req.session.user`

`SessionData` interface already contains `[key: string]: any` (see [link](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/59d4efdd3f0fbcb6f7a9cce667b9421e685ab632/types/express-session/index.d.ts#L25)), hence I've not added it in

**Comment by karrui** [comment]
> @karrui
> Re your comment in #405 that we should extend `request` to also have `req.session.user`
> 
> `SessionData` interface already contains `[key: string]: any` (see [link](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/59d4efdd3f0fbcb6f7a9cce667b9421e685ab632/types/express-session/index.d.ts#L25)), hence I've not added it in

Yea, but `any` is not very useful, and we the key is any string haha. I'm saying to overload it to specifically have user:

```ts
user?: IUserSchema
```

**Comment by shuli-ogp** [comment]
@karrui I added in user, see if ok!

File: src/app/utils/request.ts [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
+export const getRequestIp = (req: Request) => {
   return req.get('cf-connecting-ip') ?? req.ip
 }
 
-export const getTrace = (req: IRequestWithId) => {
+export const getTrace = (req: Request) => {
   return req.get('cf-ray') ?? req.id // trace using cloudflare cf-ray header, with x-request-id header as backup
 }
+
+export const getMeta = (req: Request) => {
```
Comment (karrui) [inline src/app/utils/request.ts:12]: nit: prefer a more obvious name like `createReqMeta`

File: src/app/controllers/admin-console.server.controller.js [hunk 2]
```diff
@@ -20,7 +20,7 @@ const Form = getFormModel(mongoose)
 const _ = require('lodash')
 
 const logger = require('../../config/logger').createLoggerWithLabel(module)
-const { getRequestIp, getTrace } = require('../utils/request')
```
Comment (karrui) [inline src/app/controllers/admin-console.server.controller.js:23]: `getTrace` is unused anymore right? Should remove all the unused imports in the other files too

**Review by karrui — CHANGES_REQUESTED** [review]

File: src/app/utils/request.ts [hunk 3]
```diff
@@ -1,14 +1,19 @@
 import { Request } from 'express'
+import { IncomingHttpHeaders } from 'http'
```
Comment (liangyuanruo) [inline src/app/utils/request.ts:2]: dangling import?

File: src/app/utils/request.ts (hunk 1 above)
Comment (shuli-ogp) [reply src/app/utils/request.ts]: changed accordingly

File: src/app/utils/request.ts (hunk 3 above)
Comment (shuli-ogp) [reply src/app/utils/request.ts]: removed

File: src/app/controllers/admin-console.server.controller.js (hunk 2 above)
Comment (shuli-ogp) [reply src/app/controllers/admin-console.server.controller.js:23]: edited accordingly

**Review by karrui — APPROVED** [review]
lgtm!

**Pull request merged**